deeplearning4j-core/src/test/java/org/deeplearning4j/nn/layers/ocnn/OCNNOutputLayerTest.java

```diff
@@ -0,0 +1,155 @@
+package org.deeplearning4j.nn.layers.ocnn;
+
+import org.deeplearning4j.datasets.iterator.impl.IrisDataSetIterator;
+import org.deeplearning4j.gradientcheck.GradientCheckUtil;
+import org.deeplearning4j.nn.conf.MultiLayerConfiguration;
+import org.deeplearning4j.nn.conf.NeuralNetConfiguration;
+import org.deeplearning4j.nn.multilayer.MultiLayerNetwork;
+import org.deeplearning4j.optimize.listeners.ScoreIterationListener;
+import org.deeplearning4j.util.ModelSerializer;
+import org.junit.Test;
+import org.nd4j.linalg.api.buffer.DataBuffer;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.dataset.DataSet;
+import org.nd4j.linalg.dataset.api.iterator.DataSetIterator;
+import org.nd4j.linalg.dataset.api.preprocessor.NormalizerStandardize;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.linalg.learning.config.NoOp;
+import org.nd4j.linalg.util.ArrayUtil;
+
+import java.io.File;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class OCNNOutputLayerTest {
+
+    private static final boolean PRINT_RESULTS = true;
+    private static final boolean RETURN_ON_FIRST_FAILURE = false;
+    private static final double DEFAULT_EPS = 1e-6;
+    private static final double DEFAULT_MAX_REL_ERROR = 1e-3;
+    private static final double DEFAULT_MIN_ABS_ERROR = 1e-8;
+
+    static {
+        Nd4j.setDataType(DataBuffer.Type.DOUBLE);
+    }
+
+
+    @Test
+    public void testLayer() {
+        DataSetIterator dataSetIterator = new IrisDataSetIterator(150,150);
+        DataSet ds = dataSetIterator.next();
+        NormalizerStandardize normalizerStandardize = new NormalizerStandardize();
+        normalizerStandardize.fit(dataSetIterator);
+        dataSetIterator.reset();
+        dataSetIterator.setPreProcessor(normalizerStandardize);
+
+        int numHidden = 2;
+        int nIn = 4;
+        boolean doLearningFirst = true;
+        MultiLayerConfiguration configuration = new NeuralNetConfiguration.Builder()
+                .seed(42).updater(new NoOp()).miniBatch(false)
+                .list(new  org.deeplearning4j.nn.conf.ocnn.OCNNOutputLayer.Builder().nIn(4)
+                        .nu(0.002)
+                        .nOut(2)
+                        .hiddenLayerSize(numHidden).build())
+                .build();
+        MultiLayerNetwork network = new MultiLayerNetwork(configuration);
```

Comment:
Add a DenseLayer or something below this... without it: there could be an error in the epsilon (input activation gradients) calculation, and we'd never know it.

Reply:
See update.